client/my-sites/email/add-mailboxes/index.tsx

```diff
@@ -84,14 +84,18 @@ const useAdditionalProps = ( {
 	selectedDomainName,
 	source = '',
 }: AddMailboxesProps ): AddMailboxesAdditionalProps => {
-	const selectedSite = useSelector( getSelectedSite ) as SiteData;
-	const selectedSiteId: number = selectedSite.ID;
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteId = selectedSite?.ID;
 	const domains = useSelector( ( state ) => getDomainsBySiteId( state, selectedSiteId ) );
-	const isLoadingDomains = useSelector(
-		( state ) =>
+	const isLoadingDomains = useSelector( ( state ) => {
+		if ( ! selectedSiteId ) {
+			return true;
```

Comment:
This will consider things to be loading if there is no selected site since the following two selectors require a site ID.